podpac/core/coordinates/coordinates1d.py

```diff
@@ -356,13 +356,24 @@ def select(self, bounds, return_indices=False, outer=False):
                 return self._select_full(return_indices)
 
         bounds = make_coord_value(bounds[0]), make_coord_value(bounds[1])
+        my_bounds = self.area_bounds.copy()
+
+        # If the bounds are of instance datetime64, then the comparison should happen at the lowest precision
+        if self.name == "time" and isinstance(my_bounds[0], np.datetime64) and isinstance(bounds[0], np.datetime64):
+            mine = np.timedelta64(1, my_bounds[0].dtype.name.replace("datetime64", "")[1:-1])
+            your = np.timedelta64(1, bounds[0].dtype.name.replace("datetime64", "")[1:-1])
+
+            if mine > your:
+                bounds = [b.astype(my_bounds[0].dtype) for b in bounds]
+            else:
+                my_bounds = [b.astype(bounds[0].dtype) for b in my_bounds]

```

Comment:
I think we should only do the else case.

If the coordinates are `[Oct 31, Nov 1, Nov 2]`
 - `select([Nov, Dec])` should return `[Nov 1, Nov 2]`
 - `select([Nov 1 10:00, Nov 1 12:00])` should return `[]` because Nov 1 isn't "within" 10:00 and 12:00 on Nov 1.

Reply:
I think it's helpful to also think about the primary use-case for this, which is interpolation, where the requested coordinates are `bounds` and the native coordinates are `my_bounds`. You can verify that in the DataSource eval:

```
        (
            self._requested_source_coordinates,
            self._requested_source_coordinates_index,
        ) = self.native_coordinates.intersect(coordinates, outer=True, return_indices=True)
```

When the requested coordinates are at higher resolution than the native_coordinates, you need convert `bounds` to the (lower) resolution of `my_bounds`. This is the `mine > your` case that I was arguing we should get rid of. So I guess I was wrong! More on that below.

When the native_coordinates are at higher resolution than the requested coordinates, then you need convert `my_bounds` to the (lower) resolution `bounds`. This is the `else` case. *Then the interpolation needs to average all of the native_coordinates.* This is similar to #238, which I believe is still unsolved (and in my opinion, is quite serious).

I argue that the `mine > your` case should *only* apply when `outer=True`. Here is the documentation on that:

```
        The default selection returns coordinates that are within the bounds::

            In [1]: c = ArrayCoordinates1d([0, 1, 2, 3], name='lat')

            In [2]: c.select([1.5, 2.5]).coordinates
            Out[2]: array([2.])

        The *outer* selection returns the minimal set of coordinates that contain the bounds::
        
            In [3]: c.select([1.5, 2.5], outer=True).coordinates
            Out[3]: array([1., 2., 3.])
```

I think that `2018-01-01` is not **within** `2018-01-01 12:00` and `2018-01-01 14:00`, which is the what select means when `outer=False`. But `2018-01-01` *is* in **the minimial set of coordinates that contain** `2018-01-01 12:00` and `2018-01-01 14:00`, which is what select means when `outer=True`.

Reply:
I can live with `mine > yours` when `outer == True`. 

I also agree that #238 being outstanding is quite serious, and some of this would be handled better with that in place.

Reply:
I edited my comment and added a note about what I think is the SMAP case (the else case) and why the unresolved handling of higher resolution native coordinates may not be an issue there. If I heard you correctly, there is 1 higher resolution data point `2018-01-01T12:00:00.000` available for each day, so no reduction is necessary when requesting data for a particular day `2018-01-01`.

Reply:
you got it correct. So the else case was the SMAP case.